File: src/main/java/net/atos/zac/policy/input/TaakInput.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.policy.input;

import net.atos.zac.authentication.LoggedInUser;

public class TaakInput extends UserInput {

  private final TaakData taak;

  public TaakInput(final LoggedInUser loggedInUser, final TaakData taakData) {
    super(loggedInUser);
    taak = taakData;
  }

  public TaakData getTaak() {
    return taak;
  }
}

File: src/main/kotlin/nl/info/zac/policy/input/TaakInput.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.zac.policy.input

import jakarta.json.bind.annotation.JsonbProperty
import nl.info.zac.authentication.LoggedInUser

data class TaakInput(
    val loggedInUser: LoggedInUser,

    @field:JsonbProperty("taak")
    val taakData: TaakData,

    val featureFlagPabcIntegration: Boolean
) : UserInput(
    loggedInUser = loggedInUser,
    zaaktype = taakData.zaaktype,
    featureFlagPabcIntegration = featureFlagPabcIntegration
)
